test/integration-tests/ui/PackageDependencyProvider.test.ts

```diff
@@ -13,107 +13,213 @@
+            const folder = folders.find(n => n.name === "Sources");
+            expect(folder).to.not.be.undefined;
+
+            expect(folder?.path).to.equal(`${testAssetPath("defaultPackage")}/Sources`);
+
+            const childFolders = await treeProvider.getChildren(folder);
+            const childFolder = childFolders.find(n => n.name === "PackageExe");
+            expect(childFolder).to.not.be.undefined;
+
+            expect(childFolder?.path).to.equal(
+                `${testAssetPath("defaultPackage")}/Sources/PackageExe`
+            );
+
+            const files = await treeProvider.getChildren(childFolder);
+            const file = files.find(n => n.name === "main.swift");
+            expect(file).to.not.be.undefined;
+
+            expect(file?.path).to.equal(
+                `${testAssetPath("defaultPackage")}/Sources/PackageExe/main.swift`
+            );
+        });
+
+        test("Lists remote dependency file structure", async () => {
+            const items = await treeProvider.getChildren();
+
+            const dep = items.find(n => n.name === "swift-markdown") as PackageNode;
+            expect(dep).to.not.be.undefined;
+
+            const folders = await treeProvider.getChildren(dep);
+            const folder = folders.find(n => n.name === "Sources");
+            expect(folder).to.not.be.undefined;
+
+            const path = `${testAssetPath("dependencies")}/.build/checkouts/swift-markdown`;
+            expect(folder?.path).to.equal(`${path}/Sources`);
+
+            const childFolders = await treeProvider.getChildren(folder);
+            const childFolder = childFolders.find(n => n.name === "CAtomic");
+            expect(childFolder).to.not.be.undefined;
+
+            expect(childFolder?.path).to.equal(`${path}/Sources/CAtomic`);
+
+            const files = await treeProvider.getChildren(childFolder);
+            const file = files.find(n => n.name === "CAtomic.c");
+            expect(file).to.not.be.undefined;
+
+            expect(file?.path).to.equal(`${path}/Sources/CAtomic/CAtomic.c`);
+        });
     });
 
-    suiteTeardown(() => {
-        treeProvider?.dispose();
+    suite("spm Resolve Update Contract Tests", function () {
+        let folderContext: FolderContext;
+        let workspaceContext: WorkspaceContext;
+
+        suiteSetup(async function () {
+            workspaceContext = await globalWorkspaceContextPromise;
+            await waitForNoRunningTasks();
+            folderContext = await folderContextPromise("dependencies");
+            await workspaceContext.focusFolder(folderContext);
+        });
+
+        test("Contract: spm resolve", async () => {
+            const result = await vscode.commands.executeCommand(COMMANDS.ResolveDependencies);
+            expect(result).to.be.true;
+        });
+
+        test("Contract: spm update", async () => {
+            const result = await vscode.commands.executeCommand(COMMANDS.UpdateDependencies);
```

Comment:
structurally these cases are not part of the tree provider. This and the above case, can be in separate suites. For example this one mapping to tests/commands/dependencies/useLocal.test.ts